app/helpers/sessions_helper.rb

```diff
@@ -56,4 +61,15 @@ def current_user_from_cookies
       @current_user = user
     end
   end
+
+  # Redirects to stored location (or to the default).
+  def redirect_back_or(default)
```

Comment:
The method name isn't descriptive enough (it's redirecting to provided path with default fallback of `forwarding_url ` and additionally deleting the latter from session).

How about:
```ruby
  def redirect_and_clear_session(location: session[:forwarding_url])
    redirect_to(location)
    session.delete(:forwarding_url)
  end
```